keel-actuator/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourceActuator.kt

```diff
@@ -124,7 +124,7 @@ class ResourceActuator(
   }
 
   private suspend fun ResourceHandler<*, *>.resolve(resource: Resource<out ResourceSpec>): Pair<Any, Any?> =
-    coroutineScope {
+    supervisorScope {
```

Comment:
According to the documentation, `coroutineScope`'s behavior is such that "when any child coroutine in this scope fails, this scope fails and all the rest of the children are cancelled". This is why the HTTP requests to clouddriver were being stopped mid-way (socket closed exception) -- when the UnknownBaseImage exception was thrown in one child coroutine, the other was forcefully cancelled.

I've replaced coroutineScope with [supervisorScope](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/supervisor-scope.html), where "a failure of a child does not cause this scope to fail and does not affect its other children".

Reply:
Wow! Good find.

Reply:
Nice work. Wasn't aware of that option.